validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGasLimit.java

```diff
@@ -0,0 +1,140 @@
+              .parser(value -> BLSPublicKey.fromBytesCompressed(Bytes48.fromHexString(value)))
+              .pattern("^0x[a-fA-F0-9]{96}$")
+              .example(
+                  "0x93247f2209abcacf57b75a51dafae777f9dd38bc7053d1af526f220a7489a6d3a2753e5f3e8b1cfe39b56f43611df74a")
+              .build());
+
+  private static final SerializableTypeDefinition<GetGasLimit.GetGasLimitResponse> GAS_LIMIT_DATA =
+      SerializableTypeDefinition.object(GetGasLimit.GetGasLimitResponse.class)
+          .name("GetGasLimitData")
+          .withField(
+              "gas_limit", CoreTypes.UINT64_TYPE, GetGasLimit.GetGasLimitResponse::getGasLimit)
+          .withOptionalField(
+              "pubkey", ValidatorTypes.PUBKEY_TYPE, GetGasLimit.GetGasLimitResponse::getPublicKey)
+          .build();
+
+  private static final SerializableTypeDefinition<GetGasLimit.GetGasLimitResponse> RESPONSE_TYPE =
+      SerializableTypeDefinition.object(GetGasLimit.GetGasLimitResponse.class)
+          .withField("data", GAS_LIMIT_DATA, Function.identity())
+          .build();
+
+  public GetGasLimit(Optional<BeaconProposerPreparer> beaconProposerPreparer) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("GetGasLimit")
+            .summary("Get Validator Gas Limit")
+            .withBearerAuthSecurity()
+            .tags(TAG_GAS_LIMIT)
+            .pathParam(PARAM_PUBKEY_TYPE)
+            .description(
+                "Get the execution gas limit for an individual validator. This gas limit is the one used by the validator when proposing blocks via an external builder. If no limit has been set explicitly for a key then the process-wide default will be returned.\n"
+                    + "The server may return a 400 status code if no external builder is configured.\n"
+                    + "WARNING: The gas_limit is not used on Phase0 or Altair networks.")
+            .response(SC_OK, "Success response", RESPONSE_TYPE)
+            .withAuthenticationResponses()
+            .withNotFoundResponse()
+            .build());
+    this.beaconProposerPreparer = beaconProposerPreparer;
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final BLSPublicKey publicKey = request.getPathParameter(PARAM_PUBKEY_TYPE);
+    final Optional<UInt64> maybeGasLimit =
+        this.beaconProposerPreparer.isPresent()
+            ? this.beaconProposerPreparer.get().getGasLimit(publicKey)
+            : Optional.empty();
+    if (maybeGasLimit.isEmpty()) {
+      request.respondError(SC_NOT_FOUND, "Gas limit not found");
+      return;
+    }
+    request.respondOk(new GetGasLimitResponse(maybeGasLimit.get(), Optional.of(publicKey)));
+  }
+
+  static class GetGasLimitResponse {
+    private final UInt64 gasLimit;
+    private final Optional<BLSPublicKey> publicKey;
+
+    public GetGasLimitResponse(final UInt64 gasLimit, final Optional<BLSPublicKey> publicKey) {
```

Comment:
Make the constructor take the publicKey non optional given we're always going to have it. it can wrap the optional.